abnf/odata-abnf-construction-rules.txt

```diff
@@ -560,11 +560,14 @@ complexColPathExpr = collectionPathExpr
  
 collectionPathExpr = count [ OPEN expandCountOption *( SEMI expandCountOption ) CLOSE ]
                    / filterExpr [ collectionPathExpr ]
+                   / OPEN collectionOption *( SEMI collectionOption ) CLOSE [ complexPathExpr ] 
                    / "/" anyExpr
                    / "/" allExpr
                    / "/" boundFunctionExpr
                    / "/" annotationExpr
- 
+
+collectionOption = orderby / index

```

Comment:
Why not also `filter`, `search` etc.?

Reply:
We already have `filterExpr` as a path segment, so no need to allow nesting `$filter`. Which of course would lean towards using index path segments instead of a nested `$index` query option.

The order of query options in the URL does not imply processing order, same as for "top-level" query options. `$index` must be processed last, but it can be specified anywhere within the parentheses.

Reply:
Not sure what this table is supposed to show: so far we did _not_ discuss adding `/$expand` segments or `/$orderby` segments.

Reply:
Where does the `$expand=Items/$filter(Delivered)` come from? This is not allowed, and not proposed by this PR or its Jira issue.

Reply:
That's why I striked it out. It was just an analogy with the table cell above it.

Reply:
To summarize my review: I find it strange that in the table above, three cases require the nested query option and one requires the path segment option. But that's probably better than introducing ambiguity.